src/tools/illink/src/ILLink.Shared/MessageFormat.cs

```diff
@@ -33,7 +35,24 @@ public static string FormatRequiresAttributeMismatch (bool memberHasAttribute, b
 				(true, false) => SharedStrings.DerivedRequiresMismatchMessage
 			};
 
-			return string.Format (format, args);
+			return TryFormat (format, args);
+		}
+
+		public static string TryFormat (string format, params object[] args)
+		{
+			string formattedString;
+			try
+			{
+				formattedString = string.Format(format, args);
+			}
+			catch (FormatException)
+			{
+#pragma warning disable RS1035 // Do not use APIs banned for analyzers - We just need a newline
+				formattedString = "Internal error formatting diagnostic. Please report the issue at https://aka.ms/report-illink" + Environment.NewLine
+					+ $"'{format}', " + $"[{string.Join(", ", args)}]";
+#pragma warning restore RS1035 // Do not use APIs banned for analyzers
+			}
+			return formattedString;
```

Comment:
We do resource string formatting all over the product but I haven't seen a place that would have a try/catch to transform internal errors to different internal errors. Is there more context on this?

Unhandled exceptions are often logged in various crash telemetries (not that I know of anyone actually looking at it in the past years, but in principle). This will swallow the possible telemetry.

Reply:
This string formatting caused errors since the number of arguments expected and provided don't always match for DAM warnings when Nullable<T> is the source of the value. The proper fix to avoid the exception is in https://github.com/dotnet/runtime/pull/110229, after which we could remove this try/catch. It still hits this exception in this PR because I wanted to separate the NativeAOT fix MakeGenericType fix and the DAM warning/arguments mismatch fix.

We could remove the tests and the try/catch from https://github.com/dotnet/runtime/pull/110229 before merging this PR if you'd rather avoid merging the try/catch.

Reply:
Yes, I think this PR can just be about the `HandleCallAction`, it's fine without test since we'll get the test.